components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java

```diff
@@ -258,10 +259,29 @@ public static boolean isValidCaptcha(String reCaptchaResponse) throws CaptchaExc
         try {
             try (InputStream in = entity.getContent()) {
                 JsonObject verificationResponse = new JsonParser().parse(IOUtils.toString(in)).getAsJsonObject();
-                if (verificationResponse == null || verificationResponse.get("success") == null ||
-                        !verificationResponse.get("success").getAsBoolean()) {
+                if (verificationResponse == null) {
                     throw new CaptchaClientException("reCaptcha verification failed. Please try again.");
                 }
+                boolean success = verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS) != null
+                        && verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS).getAsBoolean();
+                if (verificationResponse.get(CaptchaConstants.CAPTCHA_SCORE) != null) {
+                    double score =  verificationResponse.get(CaptchaConstants.CAPTCHA_SCORE).getAsDouble();
+                    // reCAPTCHA v3 response contains score
+                    if (log.isDebugEnabled()) {
+                        log.debug("reCAPTCHA v3 success:" + success + ", action:" +
```

Comment:
If it is reCAPTCHA v3, are relying on this success variable? If the score is less than the Threshold configured, this debug logs is misleading. (I believe this is true at this stage for all requests in v3 version.)